02_knight_and_princess/anton_gadzikovskiy/graph_generator.cpp

```diff
@@ -0,0 +1,115 @@
+#include <array>
+#include <fstream>
+#include <iostream>
+#include <string>
+#include <vector>
+using std::array;
+using std::string;
+using std::to_string;
+using std::vector;
+using VertexId = int;
+using EdgeId = int;
+constexpr int kVerticesCount = 14;
+
+struct Vertex {
+  VertexId id;
+  vector<int> edge_ids;
+};
+struct Edge {
+  EdgeId id;
+  std::pair<int, int> vertex_ids;
+};
+class Graph {
+ public:
+  vector<Vertex> vertices;
+  vector<Edge> edges;
+  void add_vertex() {
+    Vertex new_vertex;
+    new_vertex.id = vertices.size();
+    new_vertex.edge_ids = {};
+    vertices.push_back(new_vertex);

```

Comment:
Чем писать миллион строк, лучше добавить в `Vertex` конструктор. Тогда запись упростится до:
```suggestion
    vertices.emplace_back(get_new_vertex_id());
```